### ISSUE TITLE
The check of the value-set doesn't work in HAPI FHIR version 4.1.0

### ISSUE THREAD

**Issue by ycadoret95** [jamesagnew/hapi-fhir#1974]
**Describe the issue**
Hello,
We are using your HAPI FHIR v4.1.0 library, and we have encountered a problem to validate value-set with our FHIR messages.
After searching in your library, we discovered that the validation of the value-set is not implemented in this version (4.1.0), and therefore validation of the value-set does not work.

We would like to know if this problem has been resolved with the current version 5.0.2 ?

Thank you for your help,
Best regards

**Environment (please complete the following information):**
 - HAPI FHIR 4.1.0

**Comment by jamesagnew** [comment]
Hi @ycadoret95 - Can you please test your issue on our public server ( hapi.fhir.org ) and if it exists there, provide a link to a test case on that server demonstrating the issue?

**Comment by wylembars** [comment]
Hi @jamesagnew  ! 
We managed to solve our problem by updating our hapi-fhir dependency to 4.2.0. This way we managed to correctly validate on our custom StructureDefinitions and ValueSets using the Validation Support Module **PrePopulatedValidationSupport** (from R4 package in version 4.2.0).
However it seems the validation does not take display names into account ! See **PrePopulatedValidationSupport#validateCode(..)** :
`if (theCode.equals(nextExpansionCode.getCode())) {
				if (Constants.codeSystemNotNeeded(theCodeSystem) || nextExpansionCode.getSystem().equals(theCodeSystem)) {
					return new CodeValidationResult(new CodeSystem.ConceptDefinitionComponent(new CodeType(theCode)));
				}
}`
By checking out most recent version 5.0.2 I tried to figure out whether or not this check has recently been implemented as I didn't see any directly related change log. From what I saw, **PrePopulatedValidationSupport#validateCode(..)** seems not to be implemented anymore, using the default implementation from **IValidationSupport** return null. Am I missing something ? Can't **PrePopulatedValidationSupport** be used to validate custom Value-Sets anymore ? 

Thanks in advance for your response.
